chat-plugins/afk.js

```diff
@@ -0,0 +1,130 @@
+		if (!user.isAway && user.name.length > 15) return this.sendReply("Your username is too long for any kind of use of this command.");
+
+		target = target ? target.replace(/[^a-zA-Z0-9]/g, '') : 'AWAY';
+		let newName = user.name;
+		let status = parseStatus(target, true);
+		let statusLen = status.length;
+		if (statusLen > 14) return this.sendReply("Your away status should be short and to-the-point, not a dissertation on why you are away.");
+
+		if (user.isAway) {
+			let statusIdx = newName.search(/\s\-\s[\u24B6-\u24E9\u2460-\u2468\u24EA]+$/);
+			if (statusIdx > -1) newName = newName.substr(0, statusIdx);
+			if (user.name.substr(-statusLen) === status) return this.sendReply("Your away status is already set to \"" + target + "\".");
+		}
+
+		newName += ' - ' + status;
+		if (newName.length > 18) return this.sendReply("\"" + target + "\" is too long to use as your away status.");
+
+		// forcerename any possible impersonators
+		let targetUser = Users.getExact(user.userid + target);
+		if (targetUser && targetUser !== user && targetUser.name === user.name + ' - ' + target) {
+			targetUser.resetName();
+			targetUser.send("|nametaken||Your name conflicts with " + user.name + (user.name.substr(-1) === "s" ? "'" : "'s") + " new away status.");
+		}
+
+		if (user.can('lock', null, room)) this.add("");
+		user.forceRename(newName, user.registered);
+		user.updateIdentity();
+		user.isAway = true;
+	},
+
+	back: function (target, room, user) {
+		if (!user.isAway) return this.sendReply("You are not set as away.");
+		user.isAway = false;
+
+		let newName = user.name;
+		let statusIdx = newName.search(/\s\-\s[\u24B6-\u24E9\u2460-\u2468\u24EA]+$/);
+		if (statusIdx < 0) {
+			user.isAway = false;
+			if (user.can('lock', null, room)) this.add("");
+			return false;
+		}
+
+		let status = parseStatus(newName.substr(statusIdx + 3), false);
+		newName = newName.substr(0, statusIdx);
+		user.forceRename(newName, user.registered);
+		user.updateIdentity();
+		user.isAway = false;
+		if (user.can('lock', null, room)) this.add("");
+	},
+
+	afk: function (target, room, user) {
+		this.parse('/away AFK');
+	},
+
+	busy: function (target, room, user) {
+		this.parse('/away BUSY');
+	},
+
+	work: function (target, room, user) {
+		this.parse('/away WORK');
+	},
+
+	working: function (target, room, user) {
+		this.parse('/away WORKING');
+	},
+
+	eating: function (target, room, user) {
+		this.parse('/away EATING');
+	},
+
+	gaming: function (target, room, user) {
+		this.parse('/away GAMING');
+	},
+
+	sleep: function (target, room, user) {
+		this.parse('/away SLEEP');
+	},
+
+	sleeping: function (target, room, user) {
+		this.parse('/away SLEEPING');
+	},
+
+	nerd: function (target, room, user) {
+		this.parse('/away NERD');
+	},
+
+	nerding: function (target, room, user) {
+		this.parse('/away NERDING');
+	},
+
+	coding: function (target, room, user) {
+		this.parse('/away CODING');
+	},
+};
```

Comment:
All these commands may actually be implemented as aliases of ``away``, but they seem very unneeded tbh.